core/consensus/babe/src/lib.rs

```diff
@@ -629,6 +635,34 @@ impl<B: Block, C> Verifier<B> for BabeVerifier<C> where
 					auxiliary: Vec::new(),
 					fork_choice: ForkChoiceStrategy::LongestChain,
 				};
+				const NANOS_PER_SEC: u32 = 1_000_000_000;
```

Comment:
Can we move this block of code to some aux function?

Reply:
We should.  We also need to get the information to the block-authoring logic somehow ― currently, it is never used by it.  However, I could not think of a better way to do it than to stuff it in a global variable (yuck!!!!).